traitsui/qt4/tabular_editor.py

```diff
@@ -353,8 +353,10 @@ def _selected_changed(self, new):
             try:
                 selected_row = self.value.index(new)
             except Exception as e:
-                from traitsui.api import raise_to_debug
-                raise_to_debug()
+                # It's not exceptional for new to be None
+                if new is not None:
```

Comment:
I think the correct thing here is to special case `new is None`, since I think we then want the `selected_row` to be -1 (see line 366 below).  Something like:
```
selected_row = self.value.index(new) if new is not None else -1
```
(or equivalent if you don't like inline `... if ... else ...`)